zookeeper-server/src/test/java/org/apache/zookeeper/server/quorum/QuorumSSLTest.java

```diff
@@ -122,10 +124,26 @@
     private static final String PORT_UNIFICATION_DISABLED = "portUnification=false\n";
 
     private static final char[] PASSWORD = "testpass".toCharArray();
-    private static final String HOSTNAME = "localhost";
+
+    private static String HOSTNAME;
+    private static String ipAddress;
+
+    {
+        InetAddress address = null;
+        try {
+            address = InetAddress.getLocalHost();
+            HOSTNAME = address.getHostName();
+            ipAddress = address.getHostAddress();
+        } catch (UnknownHostException e) {
+            e.printStackTrace();
+        }
```

Comment:
-  When `UnknownHostException` happens, `ipAddress` should still use the `127.0.0.1` as default ?
- >  Depending on localhost and /etc/hosts config, this test can fail.. So trying make it more resilient.

  Could you plz explain more for me to understand the benefit of this change?

Reply:
@maoling , The test fails if /etc/hosts has ipv6 because localhost or ip address is hardcoded in the tests. The change makes it agostic to host setup.